akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala

```diff
@@ -745,15 +745,15 @@ private[stream] object MaterializerSession {
 /**
  * INTERNAL API
  */
-private[stream] abstract class MaterializerSession(val topLevel: StreamLayout.Module, val initialAttributes: Attributes) {
+abstract class MaterializerSession(val topLevel: StreamLayout.Module, val initialAttributes: Attributes) {
   import StreamLayout._
 
   // the contained maps store either Subscriber[Any] or VirtualPublisher, but the type system cannot express that
-  private var subscribersStack: List[ju.Map[InPort, AnyRef]] =
+  var subscribersStack: List[ju.Map[InPort, AnyRef]] =
```

Comment:
Hm, why make this public? The session should protect this internal state from being corrupted. I would like to hear the use-case here. Even if it turns out to be really needed by subclasses, protected might be better.